divviup/src/test/java/org/divviup/android/MockContext.java

```diff
@@ -0,0 +1,557 @@
+package org.divviup.android;
+
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.ContentResolver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.IntentSender;
+import android.content.ServiceConnection;
+import android.content.SharedPreferences;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.content.res.AssetManager;
+import android.content.res.Configuration;
+import android.content.res.Resources;
+import android.database.DatabaseErrorHandler;
+import android.database.sqlite.SQLiteDatabase;
+import android.graphics.Bitmap;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
+import android.os.UserHandle;
+import android.view.Display;
+
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+
+/** @noinspection NullableProblems*/
+public class MockContext extends Context {
```

Comment:
I see there is an `android.test.mock.MockContext` https://developer.android.com/reference/android/test/mock/MockContext.

> A mock [Context](https://developer.android.com/reference/android/content/Context) class. All methods are non-functional and throw [UnsupportedOperationException](https://developer.android.com/reference/java/lang/UnsupportedOperationException). You can use this to inject other dependencies, mocks, or monitors into the classes you are testing.

Can we have a `public class ActualMockContext extends MockContext` which has the new constructor and just overrides `public File getCacheDir()`? It would save a lot of boilerplate, if it works.

Reply:
That is only available to the androidTest source sets, I think, as the unit test build doesn't have `android.jar` on the classpath.

```
> Task :divviup:compileDebugUnitTestJavaWithJavac
/home/david/AndroidStudioProjects/DivviUp/divviup/src/test/java/org/divviup/android/HostSmokeTest.java:6: error: package android.test.mock does not exist
import android.test.mock.MockContext;
```

Reply:
I switched to using Mockito to implement this.